src/base.jl

```diff
@@ -666,32 +666,35 @@ end
 
 struct ServiceContributingDevices
     service::Service
-    contributing_devices::Vector{Device}
+    contributing_devices::Vector{Component}
```

Comment:
Why components here? This could enable weird behavior like having a bus or a cost function be part of a service.

Reply:
I answered myself this below.